### ISSUE TITLE
example application - KeyError code_verifier

### ISSUE THREAD

**Issue by yarox24** [mblackgeo/flask-cognito-lib#9]
Hi,

When using example application following error occurs after successful authentication:
![image](https://user-images.githubusercontent.com/18016218/176717963-76210009-8fa3-4fdd-a76f-19e88a3ba08d.png)

Testing initial URL: 
http://localhost:5000

Error line:
https://github.com/mblackgeo/flask-cognito-lib/blob/4a58b5da33f67c77a0b16764b19761623368a04a/src/flask_cognito_lib/decorators.py#L78

Variable status:
![image](https://user-images.githubusercontent.com/18016218/176718286-8a4bfc24-31cb-41d5-8f05-33a5aadb289a.png)

Libraries version:
Flask: 2.1.2
flask-cognito-lib: 1.3.0
Python 3.10 (Windows x64)

**Comment by chaseheim** [comment]
Saw the report for this pop up, wanted to add that I have also encountered this issue. In my case I did have a secret key set and still encountered the issue. You can view my auth code at https://github.com/chaseheim/WisIoHTFlaskWithDash/tree/main/application/auth using version 1.1.6.

Also occurred only after a successful authentication. Additional sign-out and sign-ins were not effected until browser restart (Browser set to reset everything on restart). My problem was reproducing it so I never reported it.

**Comment by chaseheim** [comment]
> If I understand correctly, do you mean that only the first successful authentication event is failing with this error?
Correct. 

And sorry about the hap-hazard updating of the previous comment. I will boot into my old OS and double check my versions of everything later today for you.

**Comment by chaseheim** [comment]
Just checked and it was Python 3.10 on Windows x64. The more I was thinking about it throughout the day it may have only happened on localhost, and not on AWS. Meaning it could be something with 3.10 specifically. I am having a hard time remembering if the error occurred when the app was on Elastic Beanstalk or not.

**Comment by yarox24** [comment]
Maybe it will be useful for someone but when you will set AWS callback to: localhost:5000
but you are browsing webpage via 127.0.0.1:5000
it could generate cookie problems (127.0.0.1 VS localhost domain) it may not see cookies from "other domain"

I'm still testing if this will be the last problem.

**Comment by nonpareil3** [comment]
hi, I have a similar issue as the original i.e. code_verifier. The problem happens in passing the session cookie between cognito_login() and cognito_login_callback() as printing the session within callback returns empty though it is well populated inside the login(). 
"Session within login_callback is <SecureCookieSession {}>" 
I have set a secret_key.  I have also tried session.modified =True but still no change.
Thank you, 
suneetha

**Comment by mblackgeo** [comment]
> hi, I have a similar issue as the original i.e. code_verifier. The problem happens in passing the session cookie between cognito_login() and cognito_login_callback() as printing the session within callback returns empty though it is well populated inside the login(). "Session within login_callback is <SecureCookieSession {}>" I have set a secret_key. I have also tried session.modified =True but still no change. Thank you, suneetha

Hi, can you confirm which version of the lib, python and what OS you are seeing this issue please?

**Comment by nonpareil3** [comment]
Sorry meant to add that to the orig. post :  Python3.8.8, lib is 1.3.1 and OS is mac

**Comment by mblackgeo** [comment]
> Sorry meant to add that to the orig. post : Python3.8.8, lib is 1.3.1 and OS is mac

Thanks, please could you open a new issue with full details and minimal reproducible example

**Comment by planestoner** [comment]
> Maybe it will be useful for someone but when you will set AWS callback to: localhost:5000 but you are browsing webpage via 127.0.0.1:5000 it could generate cookie problems (127.0.0.1 VS localhost domain) it may not see cookies from "other domain"
> 
> I'm still testing if this will be the last problem.

I ran into the same problem when testing locally; Cognito App Client>Hosted UI>Callback URL requires `http://localhost:5000/postlogin`, however, when calling `http://127.0.0.1:5000/postlogin` I got `Session within login_callback is <SecureCookieSession {}>`

Everything worked ok when calling `http://localhost:5000/postlogin`. I did however, have to set `app.config["AWS_COGNITO_EXPIRATION_LEEWAY"] = 5` as I got `flask_cognito_lib.exceptions.TokenVerifyError: Token is not valid`

Python 3.9.2
Version 1.6.1
